plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -4251,6 +4257,34 @@ public void rebootVM(final Connection conn, final VM vm, final String vmName) th
         }
     }
 
+    protected String skipOrRemoveSR(Connection conn, SR sr) {
+        if (sr == null) {
+            return null;
+        }
+        if (s_logger.isDebugEnabled()) {
+            s_logger.debug(logX(sr, "Removing SR"));
+        }
+        try {
+            Set<VDI> vdis = sr.getVDIs(conn);
+            for (VDI vdi : vdis) {
+                Map<java.lang.String, Types.VdiOperations> currentOperation = vdi.getCurrentOperations(conn);
+                if (currentOperation == null || currentOperation.size() == 0) {
+                    continue;
+                }
+                return null;
+            }
+            removeSR(conn, sr);
+            return null;
+        } catch (XenAPIException e) {
+            s_logger.warn(logX(sr, "Unable to get current opertions " + e.toString()), e);
+        } catch (XmlRpcException e) {
+            s_logger.warn(logX(sr, "Unable to get current opertions " + e.getMessage()), e);
+        }
+        String msg = "Remove SR failed";
+        s_logger.warn(msg);
+        return msg;
```

Comment:
"wasting time, finding out how it works"? That is definitely a contradiction for someone trying to enhance the code. relying on javadoc is a no go fomr that perspective. Reading javadoc if you want to enhance a system like this would be were the time is wasted. The exact working as it can only be seen in the code is what it is about. Therefore if there is a call for comments of any kind the code must be revisited. If we are writing public extension points for others to base their work on our binaries I would consider javadoc as an answer but still I would want it to be gnerated from the code. the code is the truth  and nothing but the code. sorry to rant on this, I have seen to much outdated comments in cloudstack.